File: solutions/0986_IntervalListIntersections.py
class Solution(object):
    def intervalIntersection(self, A, B):
        """
        :type A: List[List[int]]
        :type B: List[List[int]]
        :rtype: List[List[int]]
        """
        a_pointer = 0
        b_pointer = 0
        merged = []
        while a_pointer < len(A) and b_pointer < len(B):
            start = max(A[a_pointer][0], B[b_pointer][0])
            end = min(A[a_pointer][1],B[b_pointer][1])
            if start <= end:
                merged.append([start,end])
            if A[a_pointer][1] < B[b_pointer][1]:
                a_pointer +=1
            else:
                b_pointer +=1
        return merged

File: solutions/0986_IntervalListIntersections.js
/*
 * @param {number[][]} A
 * @param {number[][]} B
 * @return {number[][]}
 */
var intervalIntersection = function(A, B) {
    let a_pointer = 0;
    let b_pointer = 0;
    let merged = [];
    while(a_pointer < A.length && b_pointer < B.length){
        let curr_a = A[a_pointer];
        let curr_b = B[b_pointer];
        let start = Math.max(curr_a[0],curr_b[0]);
        let end = Math.min(curr_a[1],curr_b[1]);
        if(start <= end){
            merged.push([start,end]);
        }
      
        if(curr_a[1] < curr_b[1]){
            a_pointer++;
        } else {
            b_pointer++;
        } 
    }
    return merged
};